cpp/daal/src/algorithms/dtrees/gbt/regression/oneapi/gbt_regression_train_dense_default_oneapi_impl.i

```diff
@@ -690,11 +764,14 @@ services::Status RegressionTrainBatchKernelOneAPI<algorithmFPType, method>::upda
     auto & kernel = kernelUpdateResponse;
 
     {
+        DAAL_ASSERT_UNIVERSAL_BUFFER(treeOrder, int, totalRows);
+        DAAL_ASSERT_UNIVERSAL_BUFFER(response, algorithmFPType, totalRows);
+
         KernelArguments args(5);
         args.set(0, treeOrder, AccessModeIds::read);
         args.set(1, response, AccessModeIds::write);
-        args.set(2, (int)iStart);
-        args.set(3, (int)nRows);
+        args.set(2, static_cast<int32_t>(iStart));
+        args.set(3, static_cast<int32_t>(nRows));
```

Comment:
Have you checked overflow somewhere? nRows is uint32_t

Reply:
Removed cast.